slice_test.go

```diff
@@ -20,3 +22,116 @@ func ExampleUniqueInterfaces() {
 	fmt.Println(u.UniqueInterfaces([]interface{}{13, "foo", "bar", 42, 13, 43, "baz"}))
 	// Output: [13 foo bar 42 43 baz]
 }
+
+func BenchmarkUniqueStrings(b *testing.B) {
+	cases := []struct {
+		Name string
+		Data []string
+	}{
+		{"slice1", []string{"foofoo", "barbar", "foo", "bazbaz", "foo", "bar", "foobar", "baz", "foobaz"}},
+		{"slice2", bigSliceOfString()},
+	}
+
+	for _, bc := range cases {
+		b.Run(bc.Name, func(b *testing.B) {
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				u.UniqueStrings(bc.Data)
+			}
+		})
+		b.Run(bc.Name+"-parallel", func(b *testing.B) {
+			b.ResetTimer()
+			b.RunParallel(func(pb *testing.PB) {
+				for pb.Next() {
+					u.UniqueStrings(bc.Data)
+				}
+			})
+		})
+	}
+}
+
+func BenchmarkUniqueInts(b *testing.B) {
+	cases := []struct {
+		Name string
+		Data []int
+	}{
+		{"slice1", []int{4, 5, 2, 6, 7, 7, 2, 4, 2, 1, 1, 2, 3, 4}},
+		{"slice2", bigSliceOfInt()},
+	}
+
+	for _, bc := range cases {
+		b.Run(bc.Name, func(b *testing.B) {
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				u.UniqueInts(bc.Data)
+			}
+		})
+		b.Run(bc.Name+"-parallel", func(b *testing.B) {
+			b.ResetTimer()
+			b.RunParallel(func(pb *testing.PB) {
+				for pb.Next() {
+					u.UniqueInts(bc.Data)
+				}
+			})
+		})
+	}
+}
+
+func BenchmarkUniqueInterfaces(b *testing.B) {
+	cases := []struct {
+		Name string
+		Data []interface{}
+	}{
+		{"slice1", []interface{}{1, 2, 2, 1, 3, 4, "foo", "bar", "foo", "foo", "barfoo"}},
+		{"slice2", bigSliceOfInterface()},
+	}
+
+	for _, bc := range cases {
+		b.Run(bc.Name, func(b *testing.B) {
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				u.UniqueInterfaces(bc.Data)
+			}
+		})
+		b.Run(bc.Name+"-parallel", func(b *testing.B) {
+			b.ResetTimer()
+			b.RunParallel(func(pb *testing.PB) {
+				for pb.Next() {
+					u.UniqueInterfaces(bc.Data)
+				}
+			})
+		})
+	}
+}
+
+func bigSliceOfInt() []int {
+	var ns []int
+	for i := 0; i < 1000; i++ {
+		nb := rand.Intn(500)
+		ns = append(ns, nb)
+	}
+	return ns
+}
+
+func bigSliceOfString() []string {
+	var ss []string
+	for i := 0; i < 1000; i++ {
+		sn := u.RandomLetters(500)
```

Comment:
`u.RandomLetters` is very different from `rand.Intn`

---

`bigSliceOfInt` generates 1000 numbers smaller than 500 with some unique, some duplicates twice, some duplicates more than twice, and in random order; which is great

`bigSliceOfString` generates 1000 random strings with a fixed length of 500 chars each, which will result in 1000 unique values, 0 duplicates, and very long strings which is not the goal of this benchmark; you should generate something similar with `bigSliceOfInt`